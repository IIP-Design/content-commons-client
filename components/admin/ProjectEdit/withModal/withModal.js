/**
 *
 * withModal
 *
 */

import React from 'react';
import { object } from 'prop-types';

import { Modal } from 'semantic-ui-react';

const withModal = ( props, Trigger, Content, options ) => {
  const { triggerProps, contentProps } = props;

  return (
    <Modal
      { ...options }
      trigger={ <Trigger { ...triggerProps } /> }
    >
      <Modal.Content>
        <Content { ...contentProps } />
      </Modal.Content>
    </Modal>
  );
};

withModal.propTypes = {
  triggerProps: object,
  contentProps: object,
  options: object,
};

export default withModal;
