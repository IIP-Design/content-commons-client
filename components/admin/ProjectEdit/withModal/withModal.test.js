import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import withModal from './withModal';

const props = {
  triggerProps: {
    one: 'prop1',
    two: 'prop2',
  },
  contentProps: {
    one: 'prop3',
    two: 'prop4',
  },
};
const Trigger = () => <div className="trigger">Modal Trigger</div>;
const Content = () => <div className="content">Modal Content</div>;
const options = {
  one: 'option1',
  two: 'option2',
};
const Component = withModal( props, Trigger, Content, options );

describe( 'withModal()', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders the Trigger component', () => {
    const wrapper = shallow( Component );
    const trigger = wrapper.prop( 'trigger' );

    expect( trigger.type() ).toEqual( Trigger() );
    expect( trigger.props )
      .toEqual( { one: 'prop1', two: 'prop2' } );
  } );

  it( 'passes option props to Trigger element', () => {
    const wrapper = shallow( Component );
    const triggerEl = wrapper.find( '.ui.modal' );

    expect( triggerEl.prop( 'one' ) )
      .toEqual( 'option1' );
    expect( triggerEl.prop( 'two' ) )
      .toEqual( 'option2' );
  } );

  it( 'renders Content component', () => {
    const wrapper = shallow( Component );
    const content = wrapper.find( 'Content' );

    expect( content.equals( <Content one="prop3" two="prop4" /> ) )
      .toEqual( true );
  } );
} );
