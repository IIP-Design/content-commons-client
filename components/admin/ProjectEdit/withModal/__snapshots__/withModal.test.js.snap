// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withModal() renders without crashing 1`] = `
<Portal
  closeOnDocumentClick={false}
  closeOnEscape={true}
  eventPool="Modal"
  mountNode={<body />}
  onClose={[Function]}
  onMount={[Function]}
  onOpen={[Function]}
  onUnmount={[Function]}
  openOnTriggerClick={true}
  trigger={
    <Trigger
      one="prop1"
      two="prop2"
    />
  }
>
  <Ref
    innerRef={
      Object {
        "current": null,
      }
    }
  >
    <ModalDimmer
      blurring={false}
      centered={true}
      inverted={false}
      mountNode={<body />}
    >
      <Ref
        innerRef={
          Object {
            "current": null,
          }
        }
      >
        <div
          className="ui modal transition visible active"
          one="option1"
          style={Object {}}
          two="option2"
        >
          <ModalContent>
            <Content
              one="prop3"
              two="prop4"
            />
          </ModalContent>
        </div>
      </Ref>
    </ModalDimmer>
  </Ref>
</Portal>
`;
